README.md

```diff
@@ -66,20 +66,18 @@ following observations:
 - When the compilers are not able to constant-fold, the generated code is extremely similar for
   the loop-based version and the ranges-based version.
 
-## Algorithms
+## Generators:
```

Comment:
Let's reorganize the README in a different PR, since a lot of things have changed in the `version-2` branch.